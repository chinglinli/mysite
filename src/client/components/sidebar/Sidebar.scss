@import '../../css/base.scss';

.root {
  min-height: 600px;
  padding: 30px;

  .otherPart {
    margin-top: 30px;
  }

  .link {
    @extend .link;
  }
}
