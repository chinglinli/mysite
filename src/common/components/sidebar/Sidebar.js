import React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { Link } from 'react-router'

import AboutMe from '../about-me'
import RecentArticles from '../recent-articles'

import styles from './Sidebar.scss'

export default class Sidebar extends React.PureComponent {
  static PropTypes = {
    aboutMeConfig: ImmutablePropTypes.map.isRequired,
    recentArticles: ImmutablePropTypes.list
  }

  render () {
    return (
      <div className={styles.root}>
        <AboutMe config={this.props.aboutMeConfig}/>
        <div className={styles.otherPart}>
          <RecentArticles articles={this.props.recentArticles}/>
          <div className={styles.archive}>
            <Link className={styles.link} to='/articles/archives/'>所有文章列表</Link>
          </div>
        </div>
      </div>
    )
  }
}
