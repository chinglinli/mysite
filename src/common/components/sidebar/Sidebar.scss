@import '../../css/base.scss';

.root {
  background: $sub-dominant-color;
  padding: 1em;
  height: 100%;

  .link {
    @extend .link;
  }

  .otherPart {
    display: none;
  }

  @media (min-width: 1200px) {
    .otherPart {
      display: block;
    }

    .archive {
      margin-top: 30px;
    }
  }
}
