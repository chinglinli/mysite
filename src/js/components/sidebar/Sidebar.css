@value dominant-color from '../../css/colors.css';


.root {
    min-height: 600px;
    padding: 30px;
}

.otherPart {
    margin-top: 30px;
}

.link {
    color: dominant-color
}
