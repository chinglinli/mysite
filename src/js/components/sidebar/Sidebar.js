import * as React from 'react';

import AboutMe from '../about-me';
import RecentArticles from '../recent-articles';

import styles from './Sidebar.scss';


export default class Sidebar extends React.Component {
    render() {
        const articles = [
            '筆記 - 資料視覺化之理論 - 2016/04/4',
            '小喵的故事3 - 天堂篇 - 2016/02/24',
            '高中生活點滴 - 2015/11/23'
        ];

        return (
            <div className={styles.root}>
                <AboutMe/>
                <RecentArticles/>
                <div className={styles.otherPart}>
                    <div className={styles.link}>所有文章列表</div>
                </div>
            </div>
        )
    }
}
