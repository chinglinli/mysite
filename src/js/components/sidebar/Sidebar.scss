@import '../../css/colors.scss';


.root {
    min-height: 600px;
    padding: 30px;

    .otherPart {
        margin-top: 30px;
    }

    .link {
        color: $dominant-color
    }
}
