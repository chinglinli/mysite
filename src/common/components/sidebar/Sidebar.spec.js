import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import Sidebar from './Sidebar'


test('It should render Sidebar component correctly', () => {
  const props = Immutable.fromJS({
    aboutMeConfig: {
      socialLinks: [
        {name: 'A', url: 'URL 1'},
        {name: 'B', url: 'URL 2'}
      ],
      quote: '今天天氣不錯，希望明天會更好！'
    },
    recentArticles: [
      {slug: 'slug 1', title: 'title 1'},
      {slug: 'slug 2', title: 'title 2'},
      {slug: 'slug 3', title: 'title 3'}
    ]
  })

  const component = renderer.create(
    <Sidebar
      aboutMeConfig={props.get('aboutMeConfig')}
      recentArticles={props.get('recentArticles')}
    />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
