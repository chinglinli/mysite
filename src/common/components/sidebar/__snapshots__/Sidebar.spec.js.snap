// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Sidebar component correctly 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <div
      className="profile"
    >
      <div
        className="author"
      />
      <div
        className="info"
      >
        <div>
          <div>
            我是一隻
            <em>
              兩大類
            </em>
          </div>
          <div>
            <a
              className="link"
              href="URL 1"
            >
              A
            </a>
            <a
              className="link"
              href="URL 2"
            >
              B
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="quote"
    >
      今天天氣不錯，希望明天會更好！
    </div>
  </div>
  <div
    className="otherPart"
  >
    <div
      className="root"
    >
      <div
        className="title"
      >
        近期文章
      </div>
      <div
        className="hr"
      />
      <div
        className="articles"
      >
        <a
          className="article"
          onClick={[Function]}
          style={Object {}}
        >
          title 1
        </a>
        <a
          className="article"
          onClick={[Function]}
          style={Object {}}
        >
          title 2
        </a>
        <a
          className="article"
          onClick={[Function]}
          style={Object {}}
        >
          title 3
        </a>
      </div>
      <div
        className="hr"
      />
    </div>
    <div
      className="archive"
    >
      <a
        className="link"
        onClick={[Function]}
        style={Object {}}
      >
        所有文章列表
      </a>
    </div>
  </div>
</div>
`;
